### ISSUE TITLE
Add cron schedule support to the `bloblang` input

### ISSUE THREAD

**Issue by Jeffail** [Jeffail/benthos#505]
The [bloblang input](https://www.benthos.dev/docs/components/inputs/bloblang) currently supports an interval field, allowing you to generate data and trigger pipelines at a given time interval. However, it would be nice to expand this to support proper cron scheduling, allowing users to either specify a static `interval`, a `cron` schedule, or neither (generate as fast as we can).

The name of this input is also not particularly useful as users are more likely to be searching either for a scheduled input or a generated input type, not necessarily a bloblang mapping input type. We should consider renaming it to something like `generate`.

Edit: Useful library giving us most of what we need: https://godoc.org/github.com/robfig/cron#hdr-Alternative_Formats

**Comment by Jeffail** [comment]
Once this is done we should also consider expanding the cron input concept by allowing users to specify a cron schedule for any child input, similar to how the `read_until` input works. The main challenge with this is that most inputs aren't designed to indefinitely block to a schedule and we might find that many inputs lose their connections during the waiting period.

**Comment by dilipkk-foyernet** [comment]
@Jeffail currently bloblang **interval** is kind blocks the ReadWithContext, it should ok for smaller intervals but if the interval is bigger then it gets blocked several hours right? I am checking this issue to see if I can contribute to this so I want to know if understanding right or not.

**Comment by Jeffail** [comment]
Hey @dilipkk-foyernet, yeah exactly, blocking `ReadWithContext` for a long time (hours, even days) is fine as long as you return `types.ErrTimeout` when `case <-ctx.Done():` unblocks.

**Comment by dilipkk-foyernet** [comment]
When we implement cron based schedule there shouldn't be schedules missed due to process restart etc otherwise it will be incomplete. I guess that would be a pretty tough problem right? we need to maintain state/history or some kind of distributed scheduler is needed with shared configuration and bookkeeping so that there is no single point of failure and then there is less chance of missing schedules. @Jeffail what do you think about this?

**Comment by Jeffail** [comment]
Hey @dilipkk-foyernet, for that we'd need coordination between benthos instances which I would say is out of scope. It's still possible to add resiliency with the scheduling by adding a queue system in between, so multiple redundant benthos instances writing cron events into something like kafka and then a layer of benthos consumers reading the topic with deduplication of the redundant trigger events.

**Comment by dilipkk-foyernet** [comment]
Thanks for your response. I will look into it and get back to you as soon
as possible.

**Comment by Jeffail** [comment]
@dilipkk-foyernet, no rush but are you still planning to work on this? Otherwise there's a chance I might tackle this whilst I'm doing some related work.

**Comment by dilipkk-foyernet** [comment]
Thanks @Jeffail/benthos <benthos@noreply.github.com>  I couldn't spend much
time on it due to other works.
I won't able to spend time on this now but in the coming month I can work
on this.

**Comment by dilipkk-foyernet** [comment]
@Jeffail how to update the documentation: https://www.benthos.dev/docs/components/inputs/bloblang

**Comment by dilipkk-foyernet** [comment]
https://github.com/Jeffail/benthos/blob/master/website/docs/components/inputs/bloblang.md
we need to update here right?

**Comment by dilipkk-foyernet** [comment]
Looks like documentation is already updated. @Jeffail thanks :+1:

**Comment by dilipkk-foyernet** [comment]
We can close this issue :)
